src/test/scala/io/iohk/ethereum/vm/OpCodeFunSpec.scala

```diff
@@ -251,28 +253,40 @@ class OpCodeFunSpec extends FunSuite with OpCodeTesting with Matchers with Prope
     }
   }
 
-  ignore("BLOCKHASH") {
-    // to be implemented
-  }
+  test(BLOCKHASH) { op =>
+    val stateGen: Gen[PS] = for {
+      extCode <- getByteStringGen(0, 256)
 
-  ignore("COINBASE") {
-    // to be implemented
-  }
+      stateIn <- getProgramStateGen(
+        stackGen = getStackGen(maxWord = DataWord(256)),
+        memGen = getMemoryGen(256),
+        codeGen = getByteStringGen(0, 256)
+      )
 
-  ignore("TIMESTAMP") {
-    // to be implemented
-  }
+      doSave <- Gen.oneOf(false, true, true)
 
-  ignore("NUMBER") {
-    // to be implemented
-  }
+      addr = Address(stateIn.stack.pop._1)
+      hash = kec256(extCode)
+      world = if (doSave) stateIn.world.saveAccount(addr, Account.Empty.copy(codeHash = hash)) else stateIn.world
+    } yield stateIn.withWorld(world)
```

Comment:
Those that do make sense. But code, memory and external accounts have nothing to do with functionality of `BLOCKHASH`. You could however add a generator for the block number, even if you're just going to repeat the defualts, to make it explicit.